torch/nn/functional.py

```diff
@@ -4841,28 +4841,79 @@ def _in_projection(
 
 scaled_dot_product_attention = _add_docstr(
     torch._C._nn.scaled_dot_product_attention, r"""
+scaled_dot_product_attention(query, key, value, attn_mask=None, dropout_p=0.0, is_causal=False) -> Tensor:
+
 Computes scaled dot product attention on query, key and value tensors, using
 an optional attention mask if passed, and applying dropout if a probability
 greater than 0.0 is specified.
 
+.. code-block:: python
+
+    # Composite Implementation
+    attn_mask = torch.ones(L, S, dtype=torch.bool).tril(diagonal=0) if is_causal else attn_mask
+    attn_mask = attn_mask.masked_fill(not attn_mask, -float('inf')) if attn_mask.dtype==torch.bool else attn_mask
+    attn_weight = torch.softmax((Q @ K.transpose(-2, -1) / math.sqrt(Q.size(-1))) + attn_mask, dim=-1)
+    attn_weight = torch.dropout(attn_weight, dropout_p)
+    return attn_weight @ V
+
+.. warning:: This function is beta and subject to change.
+
+Note:
+    This function calls into one of three backends:
+        * `FlashAttention: Fast and Memory-Efficient Exact Attention with IO-Awareness`_
+        * `Memory-Efficient Attention`_
+        * A pytorch implementation defined in c++ matching the above mathematical formulation
+
+    The function defaults to selecting the highest-performing implementation based on the inputs provided.
+    However, each of the fused kernels has specific input limitations.
+    If you require a specific backend to be utilized, there exists functions to enable or disable specific backends.
+    Please note that all backends are enabled by default.
+
+    For example :func:`~torch.backends.cuda.enable_flash_sdp` can be used to enable/disable FlashAttention.
```

Comment:
I would remove the `~` to get the full path here. Since the function is not in the current namespace, it makes it easier to find it.